<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Ying Shan, Daniel E. Weeks" />


<title>BayesRB-vignette</title>

<script>// Hide empty <a> tag within highlighted CodeBlock for screen reader accessibility (see https://github.com/jgm/pandoc/issues/6352#issuecomment-626106786) -->
// v0.0.1
// Written by JooYoung Seo (jooyoung@psu.edu) and Atsushi Yasumoto on June 1st, 2020.

document.addEventListener('DOMContentLoaded', function() {
  const codeList = document.getElementsByClassName("sourceCode");
  for (var i = 0; i < codeList.length; i++) {
    var linkList = codeList[i].getElementsByTagName('a');
    for (var j = 0; j < linkList.length; j++) {
      if (linkList[j].innerHTML === "") {
        linkList[j].setAttribute('aria-hidden', 'true');
      }
    }
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>






<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">BayesRB-vignette</h1>
<h4 class="author">Ying Shan, Daniel E. Weeks</h4>
<h4 class="date">二月 21, 2022</h4>


<div id="TOC">
<ul>
<li><a href="#overview"><span class="toc-section-number">1</span> Overview</a></li>
<li><a href="#installation"><span class="toc-section-number">2</span> Installation</a></li>
<li><a href="#data-format"><span class="toc-section-number">3</span> Data Format</a><ul>
<li><a href="#the-input-data"><span class="toc-section-number">3.1</span> The input data</a></li>
<li><a href="#the-genotype-file"><span class="toc-section-number">3.2</span> The genotype file</a></li>
<li><a href="#the-phenotype-file"><span class="toc-section-number">3.3</span> The phenotype file</a></li>
</ul></li>
<li><a href="#how-to-run-the-program"><span class="toc-section-number">4</span> How to Run the Program</a></li>
<li><a href="#explanation-of-the-results"><span class="toc-section-number">5</span> Explanation of the Results</a></li>
</ul>
</div>

<p>Copyright 2022, Peking University Shenzhen Hospital and University of Pittsburgh. All Rights Reserved.</p>
<div id="overview" class="section level1">
<h1><span class="header-section-number">1</span> Overview</h1>
<p>BayesRB is a novel MCMC based polygenic genetic risk score algorithm for dichotomous traits. The R package under the same name is for SNP effect estimation and genetic risk prediction for dichotomous traits. BayesRB is an extension of BayesR proposed by Moser et al. BayesR method performs well on the SNP effect estimation and genetic risk prediction, but it is designed to be applied to the data with quantitative traits. BayesRB allows the dichotomous outcomes. It inherited the characteristics of unbiasedness, accuracy, sparseness, robustness and powerfulness.</p>
<p>BayesRB R package is written by Rcpp.</p>
</div>
<div id="installation" class="section level1">
<h1><span class="header-section-number">2</span> Installation</h1>
<p>The package is written in R language. To install, proceed as follows:</p>
<ol style="list-style-type: decimal">
<li>Install the <code>devtools</code> package by starting up R and issuing this command:</li>
</ol>
<pre><code>install.packages(&quot;devtools&quot;)</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Load the <code>devtools</code> library to make its commands available:</li>
</ol>
<pre><code>library(devtools)</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>Install the <code>BayesRB</code> R package from the github repository via this command:</li>
</ol>
<pre><code>install_github(&quot;sylviashanboo/bayesRB&quot;)</code></pre>
<ol start="4" style="list-style-type: decimal">
<li>After the <code>BayesRB</code> R package has been installed, you can view this vignette by issuing these commands at the R prompt:</li>
</ol>
<pre><code>library(BayesRB)
browseVignettes(&quot;BayesRB&quot;)</code></pre>
</div>
<div id="data-format" class="section level1">
<h1><span class="header-section-number">3</span> Data Format</h1>
<div id="the-input-data" class="section level2">
<h2><span class="header-section-number">3.1</span> The input data</h2>
<ul>
<li>seed: the seed set for MCMC loop. It should be an integer.</li>
<li>MCMC_iteration: the number of MCMC iterations. It should be an integer.</li>
<li>burn_iteration: the number of interations as warming up. It should be an integer.</li>
<li>thin_value: thin value. It should be an integer.</li>
<li>X: the genotype file. It is an N rows x P columns matrix with N samples and P SNPs.</li>
<li>Y: the phenotype file. It is a vector of length N. Again N is the sample size. Y = 1 indicates a case, otherwise a control.</li>
<li>beta_initial: a vector with length p+1, containing the initial values of the grand mean and the SNP effects.The initial value of the grand mean goes first, followed by the initial values of the SNP effects. SNP order stays the same as the columns in the X matrix.</li>
</ul>
</div>
<div id="the-genotype-file" class="section level2">
<h2><span class="header-section-number">3.2</span> The genotype file</h2>
<p>The X matrix described above is the genotype file. It is an n rows x p columns matrix with N samples and P SNPs. For each SNP, the numbers of risk alleles are standardized as Norm(0,1).Below is an example of the genotype file.</p>
<table>
<caption>Table 1: The first 6 rows and first 8 columns of the example X matrix</caption>
<colgroup>
<col width="12%"></col>
<col width="12%"></col>
<col width="12%"></col>
<col width="12%"></col>
<col width="12%"></col>
<col width="12%"></col>
<col width="12%"></col>
<col width="12%"></col>
</colgroup>
<thead>
<tr class="header">
<th align="right">X1</th>
<th align="right">X2</th>
<th align="right">X3</th>
<th align="right">X4</th>
<th align="right">X5</th>
<th align="right">X6</th>
<th align="right">X7</th>
<th align="right">X8</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">0.5750507</td>
<td align="right">-0.8952397</td>
<td align="right">-0.8976062</td>
<td align="right">0.9992161</td>
<td align="right">-1.4349886</td>
<td align="right">-0.7345784</td>
<td align="right">2.0919237</td>
<td align="right">0.8139142</td>
</tr>
<tr class="even">
<td align="right">0.5750507</td>
<td align="right">-0.8952397</td>
<td align="right">0.6661677</td>
<td align="right">0.9992161</td>
<td align="right">0.5776181</td>
<td align="right">-0.7345784</td>
<td align="right">-0.9137138</td>
<td align="right">0.8139142</td>
</tr>
<tr class="odd">
<td align="right">-1.4860629</td>
<td align="right">-0.8952397</td>
<td align="right">2.2299416</td>
<td align="right">-0.4447378</td>
<td align="right">0.5776181</td>
<td align="right">-0.7345784</td>
<td align="right">-0.9137138</td>
<td align="right">0.8139142</td>
</tr>
<tr class="even">
<td align="right">0.5750507</td>
<td align="right">0.7781055</td>
<td align="right">2.2299416</td>
<td align="right">-0.4447378</td>
<td align="right">0.5776181</td>
<td align="right">-0.7345784</td>
<td align="right">-0.9137138</td>
<td align="right">0.8139142</td>
</tr>
<tr class="odd">
<td align="right">0.5750507</td>
<td align="right">0.7781055</td>
<td align="right">-0.8976062</td>
<td align="right">-0.4447378</td>
<td align="right">0.5776181</td>
<td align="right">-0.7345784</td>
<td align="right">0.5891049</td>
<td align="right">-0.8204516</td>
</tr>
<tr class="even">
<td align="right">-1.4860629</td>
<td align="right">-0.8952397</td>
<td align="right">-0.8976062</td>
<td align="right">-1.8886918</td>
<td align="right">0.5776181</td>
<td align="right">-0.7345784</td>
<td align="right">-0.9137138</td>
<td align="right">0.8139142</td>
</tr>
</tbody>
</table>
</div>
<div id="the-phenotype-file" class="section level2">
<h2><span class="header-section-number">3.3</span> The phenotype file</h2>
<p>The Y matrix described above is the phenotype file. It’s a vector with N elements, which indicate the disease status (0,1) of the N samples. Below is an example of the phenotype file.</p>
<pre><code>## Warning in kable_pipe(x = structure(c(&quot;0&quot;, &quot;1&quot;, &quot;1&quot;, &quot;0&quot;, &quot;0&quot;, &quot;0&quot;, &quot;0&quot;, : The
## table should have a header (column names)</code></pre>
<table>
<caption>Table 2: The first 20 elements of the example Y matrix</caption>
<tbody>
<tr class="odd">
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="how-to-run-the-program" class="section level1">
<h1><span class="header-section-number">4</span> How to Run the Program</h1>
<p>The main functions in this package which implement the statistics described in Shan et al. (2022, 2016):</p>
<p><code>BayesRB</code></p>
<p>After installing the <code>BayesRB</code> R package, you can access BayesRB help page easily, which also contains example code.</p>
<pre><code>library(BayesRB)
?BayesRB</code></pre>
</div>
<div id="explanation-of-the-results" class="section level1">
<h1><span class="header-section-number">5</span> Explanation of the Results</h1>
<p>##The output parameters The results from BayesRB function are the estimated parameters in each MCMC loop after burn-in and thin (valid MCMC loop):</p>
<ul>
<li>r_beta: the estimated coefficients of all the SNPs in the valid MCMC loop. The matrix contains I rows and P columns, where I is the number of the valid MCMC loops, and P is the number of SNPs.</li>
<li>r_bj: the variance groups (1-4) to which the SNPs are estimated to belong in each valid MCMC loop. The matrix contains I rows and P columns, where I is the number of the valid MCMC loops, and P is the number of SNPs.</li>
<li>r_lambda: the estimated lambda parameter for each sample in each valid MCMC loop.The matrix contains I rows and N columns, where I is the number of the valid MCMC loops, and N is the number of samples.</li>
<li>r_Z: the estimated auxiliary variable Z for each sample in each valid MCMC loop.The matrix contains I rows and N columns, where I is the number of the valid MCMC loops, and N is the number of samples.</li>
<li>r_mu: the estimated grand mean variable mu in each valid MCMC loop.</li>
<li>r_sigma2: the estimated genetic variance sigma2 in each valid MCMC loop.</li>
<li>r_pi: the estimated mixture proportions pi1-pi4 in each valid MCMC loop. The matrix contains I rows and 4 columns, where I is the number of the valid MCMC loops.</li>
<li>r_m: the estimated SNPs being in the category 1-4 in each valid MCMC loop. The matrix contains I rows and 4 columns, where I is the number of the valid MCMC loops.</li>
</ul>
<p>##The calculation of PRS and probability of being a case With the estimated parameters provided by BayesRB, PRS and probability of being a case can be easily calculated:</p>
<p>PRS:</p>
<pre><code>load(&quot;../data/result.Rdata&quot;)
PRS = mean(result$r_mu) + x%*%apply(result$r_beta,2,mean)
hist(PRS)</code></pre>
<p>The probability of being a case:</p>
<pre><code>load(&quot;../data/result.Rdata&quot;)
prs_hat = apply(est_Z,1,ilogit)
hist(prs_hat)</code></pre>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
